### PR TITLE
AWS: resolve subnet-related FIXMEs

### PR DESCRIPTION
**Pull request by r4f4**
When looking at the subnet code I noticed the comment `// FIXME: port to DescribeSubnetsPagesWithContext once we bump our vendored AWS package past v1.19.30`. So I did it.

### CODE REVIEW

File: pkg/destroy/aws/aws.go [hunk 1]
```diff
@@ -1308,26 +1308,27 @@ func deleteEC2SubnetsByVPC(ctx context.Context, client *ec2.EC2, vpc string, fai
 				},
 			},
 		},
+		func(results *ec2.DescribeSubnetsOutput, lastPage bool) bool {
+			for _, subnet := range results.Subnets {
+				err := deleteEC2Subnet(ctx, client, *subnet.SubnetId, logger.WithField("subnet", *subnet.SubnetId))
+				if err != nil {
+					err = errors.Wrapf(err, "deleting EC2 subnet %s", *subnet.SubnetId)
+					if lastError != nil {
```
Comment (barbacbd) [inline pkg/destroy/aws/aws.go:1316]: This could print the same error multiple times. Is this intended or should this only print new errors?

**Review by barbacbd — COMMENTED** [review]
/lgtm

File: pkg/destroy/aws/aws.go (hunk 1 above)
Comment (r4f4) [reply pkg/destroy/aws/aws.go:1316]: We discussed this on Slack and we don't think the error will be printed multiple times. Also, this was the existent behaviour already.

**Pull request merged**